crates/inverted/src/lib.rs

```diff
@@ -0,0 +1,162 @@
+#![allow(clippy::len_without_is_empty)]
+
+pub mod operator;
+
+use self::operator::OperatorInvertedSparse;
+use base::index::{IndexOptions, SearchOptions};
+use base::operator::Borrowed;
+use base::scalar::{ScalarLike, F32};
+use base::search::{Collection, Element, Payload, Source, Vectors};
+use common::dir_ops::sync_dir;
+use common::json::Json;
+use common::mmap_array::MmapArray;
+use common::remap::RemappedCollection;
+use storage::Storage;
+
+use std::collections::{BTreeMap, BinaryHeap};
+use std::fs::create_dir;
+use std::path::Path;
+
+const ZERO: F32 = F32(0.0);
+
+#[allow(dead_code)]
+pub struct InvertedSparse<O: OperatorInvertedSparse> {
+    storage: O::Storage,
+    payloads: MmapArray<Payload>,
+    indexes: Json<Vec<u32>>,
+    offsets: Json<Vec<u32>>,
+    scores: Json<Vec<F32>>,
+}
+
+impl<O: OperatorInvertedSparse> InvertedSparse<O> {
+    pub fn create(path: impl AsRef<Path>, options: IndexOptions, source: &impl Source<O>) -> Self {
+        let remapped = RemappedCollection::from_source(source);
+        from_nothing(path, options, &remapped)
+    }
+
+    pub fn open(path: impl AsRef<Path>) -> Self {
+        open(path)
+    }
+
+    pub fn vbase<'a>(
+        &'a self,
+        vector: Borrowed<'a, O>,
+        _: &'a SearchOptions,
+    ) -> (Vec<Element>, Box<(dyn Iterator<Item = Element> + 'a)>) {
+        let mut doc_score = vec![ZERO; self.payloads.len()];
+        for (token, val) in O::to_index_vec(vector) {
+            let start = self.offsets[token as usize];
+            let end = self.offsets[token as usize + 1];
+            for i in (start as usize)..(end as usize) {
+                doc_score[self.indexes[i] as usize] += self.scores[i] * val;
+            }
+        }
+        let mut candidates: BinaryHeap<(F32, Payload)> = doc_score
+            .iter()
+            .enumerate()
+            .filter(|&(_, score)| *score > ZERO)
```

Comment:
Why?

Reply:
No matter if it's meaningful, it should be returned.

Reply:
Why should we return the documents that are not relevant at all?

Reply:
A query should return all tuples in order if there are only `ORDER BY` clauses according to semantics of SQL.